capsules/src/virtual_alarm.rs

```diff
@@ -143,31 +139,32 @@ impl<'a, Alrm: Alarm> time::Client for MuxAlarm<'a, Alrm> {
 
         // Check whether to fire each alarm. At this level, alarms are one-shot,
         // so a repeating client will set it again in the fired() callback.
-        for cur in self.virtual_alarms.iter() {
-            let should_fire = past_from_base(cur.when.get(), now + 100, prev);
-            if cur.armed.get() && should_fire {
+        self.virtual_alarms
+            .iter()
+            .filter(|cur| cur.armed.get() && has_expired(cur.when.get(), now, prev))
```

Comment:
There used to be a magic 100 in the check here. Is that slop no longer necessary?

Reply:
Correct, that magic 100 was a hack that just seemed to work and I believe it is now unnecessary because we're explicitly checking for the case where the virtual alarm expires between HW alarm firing and resetting the new alarm.